endaq/calc/stats.py

```diff
@@ -6,6 +6,342 @@
+        metrics = pd.concat([metrics,peak,rms])
+
+    #Optionally Calculate PVSS
+    if include_pseudo_velocity:
+        freqs = utils.logfreqs(df, bins_per_octave=bins_per_octave, init_freq=init_freq)
+        pvss = shock.shock_spectrum(df, freqs=freqs, damp=damp, mode='pvss')*vel_disp_multiplier
+        if include_resultant:
+            pvss['Resultant'] = shock.shock_spectrum(df, freqs=freqs, damp=damp, mode='pvss', aggregate_axes=True)['resultant']*vel_disp_multiplier
+
+        if display_plots:
+            px.line(pvss,log_x=True,log_y=True).update_layout(yaxis_title_text = 'Pseudo Velocity',xaxis_title_text='Natural Frequency (Hz)').show()
+
+        pvss = pd.DataFrame(pvss.max()).reset_index()
+        pvss.columns = ['variable','value']
+        pvss['calculation'] = 'Peak Pseudo Velocity'
+        metrics = pd.concat([metrics,pvss])
+
+    #Return Metrics
+    return metrics
+
+
+def find_peaks(
+        df: pd.DataFrame,
+        time_distance: float = 1.0,
+        add_resultant: bool = False,
+        threshold: float = None,
+        threshold_reference: typing.Literal["rms", "peak"] = "rms",
+        threshold_multiplier: float = 4.0,
+        use_abs: bool = True,
+        display_plots: bool = False,
+) -> pd.DataFrame:
+    """
+    Find the peak events of a given time series using the maximum across all input columns
+    :param df: the input dataframe with an index defining the time in seconds or datetime
+    :param time_distance: the minimum time in seconds between events, default is 1.0
+    :param add_resultant: add a resultant (root sum of the squares) to the time series prior to finding peaks, calculated from the other input dataframe columns
+    :param threshold: if `None` (default) this is ignored, but if defined this value is passed as the minimum threshold to define a shock event
+    :param threshold_reference: if the threshold isn't defined, calculate it from:
+        - `"rms"` (default) the RMS value of the time series
+        - `"peak"` the overall peak
+    :param threshold_multiplier: if the threshold isn't defined, multiply this by the `threshold_reference`, suggestions are:
+        - `4.0` (default) when using RMS, a typical Gaussian signal has a kurtosis of 3.0
+        - `0.1` when using peak, to get all events greater than 10% of the overall peak
+    :param use_abs: use the absolute value of the data to define peak events, default is True
+    :param display_plots: display a plotly figure of the time series with the peak events plotted over it (default as False)
+    :return: an array of index locations
+
+    .. seealso::
+        - `SciPy find_peaks function <https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.find_peaks.html>`_
+          Documentation for the SciPy function used to identify the peak events.
+
+    Here's an example implementation using a 60M dataset that loads the data, finds the peaks, and plots in under 10 seconds
+    .. code:: python
```

Comment:
This isn't rendering right, I think you need an extra empty space between lines with a directive

```suggestion
    Here's an example implementation using a 60M dataset that loads the data, finds the peaks, and plots in under 10 seconds
    
    .. code:: python
```